src/base.jl

```diff
@@ -232,6 +250,26 @@ for T in (UInt2048, UInt1024, UInt512, UInt256, UInt128, UInt64, UInt32)
     end
 end
 
+"""
+Return a ShortString type that can hold maxlen codeunits
+The keyword parameter `types` can be used to pass a list of types
+which can be used to store the string
+If no type is large enough, then an `ArgumentError` is thrown
+"""
+function get_type(maxlen; types=def_types)
+    for T in types
+        maxlen <= sizeof(T) - size_bytes(T) && return ShortString{T}
+    end
+    throw(ArgumentError("$maxlen is too large to fit into any of the provided types: $types"))
+end
+
+ShortString(str::Union{String,SubString{String}}, maxlen = 0; types=def_types) =
+    get_type(maxlen <= 0 ? sizeof(str) : maxlen, types=types)(str)
+
+macro ss_str(str, max="0")
```

Comment:
Needs a docstring, and a compat note?
Do we want to add the ability to accept `x7` etc for compat with older julia version?

Reply:
The other string macros haven't been removed (although I'd like to).
Same thing with all of the type alias `ShortString3`...`ShortString255`.
`ShortString(str, 255)` I think is just as easy to use.

Reply:
I'll add the docstring and the compat note in the next round, OK?